integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java

```diff
@@ -0,0 +1,692 @@
+import org.apache.camel.quarkus.component.dataformats.json.model.Order;
+import org.apache.camel.quarkus.component.dataformats.json.model.Pojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestJAXBPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestOtherPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestPojo;
+import org.apache.camel.quarkus.component.dataformats.json.model.TestPojoView;
+import org.apache.camel.support.DefaultExchange;
+import org.jboss.logging.Logger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@Path("/dataformats-json")
+@ApplicationScoped
+public class JacksonJsonResource {
+
+    private static final Logger LOG = Logger.getLogger(JacksonJsonResource.class);
+    private static final String LS = System.lineSeparator();
+    @Inject
+    ProducerTemplate producerTemplate;
+    @Inject
+    ConsumerTemplate consumerTemplate;
+    @Inject
+    CamelContext context;
+
+    @Path("jackson/unmarshal-typeheader")
+    @GET
+    public void jacksonXmlUnmarshalTypeHeader(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-type-header-reversePojo", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(TestPojo.class);
+
+        String json = "{\"name\":\"Camel\"}";
+        producerTemplate.sendBodyAndHeader("direct:jackson-unmarshal-type-header-backPojo", json,
+                JacksonConstants.UNMARSHAL_TYPE, TestPojo.class.getName());
+
+        mock.assertIsSatisfied();
+
+        TestPojo pojo = mock.getReceivedExchanges().get(0).getIn().getBody(TestPojo.class);
+        assertNotNull(pojo);
+        assertEquals("Camel", pojo.getName());
+    }
+
+    @Path("jackson/unmarshal-list")
+    @GET
+    public void jacksonUnmarshalList(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-reversePojo", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(List.class);
+
+        String json = "[{\"name\":\"Camel\"}, {\"name\":\"World\"}]";
+        producerTemplate.sendBody("direct:jackson-unmarshal-backPojo", json);
+
+        mock.assertIsSatisfied();
+
+        List list = mock.getReceivedExchanges().get(0).getIn().getBody(List.class);
+        assertNotNull(list);
+        assertEquals(2, list.size());
+
+        TestPojo pojo = (TestPojo) list.get(0);
+        assertEquals("Camel", pojo.getName());
+        pojo = (TestPojo) list.get(1);
+        assertEquals("World", pojo.getName());
+
+        mock.expectedMessageCount(2);
+        mock.message(1).body().isInstanceOf(List.class);
+
+        json = "[{\"name\":\"Camel\"}]";
```

Comment:
I wonder whether testing unmarshlling a single element bring some value in the context of camel-quarkus ? We are are already testing with 2 elements and I don't think there is anything special related to quarkus or graalvm ?

Reply:
Yes, this isn't quarkus specific, I just simply ported it from camel. And I think more test is good then less here.